limiter_test.go

```diff
@@ -55,3 +55,35 @@ func TestLimit(t *testing.T) {
 		t.Error("invalid max", max)
 	}
 }
+
+func TestExecuteWithParams(t *testing.T) {
+	LIMIT := 2
+	N := 30
+	lock := &sync.Mutex{}
+	validatorCounter := 0
+
+	c := NewConcurrencyLimiter(LIMIT)
+	for i := 0; i < N; i++ {
+		validatorCounter += i
+
+		c.ExecuteWithParams(func(params ...interface{}) {
```

Comment:
by adding this function, you are encouraging less strict typing of the API. are we sure we would like to go that way?
why can't the context of the anonymous function passed to 'Execute' capture all the parameters it needs?
lambdas are super cheap in in Go lang ...